### ISSUE TITLE
Protobuf.Encoder.fix_value

### ISSUE THREAD

**Issue by scottmessinger** [pryin-io/pryin#14]
After install Pryin and hit my API, I encounter the following error.  I'm using Pryin 1.2.0

```
backend  │ [error] GenServer PryIn.InteractionForwarder terminating
backend  │ ** (FunctionClauseError) no function clause matching in Enum."-map/2-lists^map/1-0-"/2
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"(#Function<2.26820709/1 in Protobuf.Encoder.fix_value/1>, "\"role_grants\"")
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
backend  │     (exprotobuf) lib/exprotobuf/encoder.ex:31: anonymous fn/2 in Protobuf.Encoder.fix_undefined/1
backend  │     (elixir) lib/enum.ex:1755: Enum."-reduce/3-lists^foldl/2-0-"/3
backend  │     (exprotobuf) lib/exprotobuf/encoder.ex:42: Protobuf.Encoder.fix_value/1
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
backend  │     (exprotobuf) lib/exprotobuf/encoder.ex:31: anonymous fn/2 in Protobuf.Encoder.fix_undefined/1
backend  │     (elixir) lib/enum.ex:1755: Enum."-reduce/3-lists^foldl/2-0-"/3
backend  │     (exprotobuf) lib/exprotobuf/encoder.ex:42: Protobuf.Encoder.fix_value/1
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
backend  │     (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
backend  │     (exprotobuf) lib/exprotobuf/encoder.ex:31: anonymous fn/2 in Protobuf.Encoder.fix_undefined/1
backend  │     (elixir) lib/enum.ex:1755: Enum."-reduce/3-lists^foldl/2-0-"/3
backend  │     (exprotobuf) lib/exprotobuf/encoder.ex:22: Protobuf.Encoder.encode/2
backend  │     (pryin) lib/pryin/interaction_forwarder.ex:33: PryIn.InteractionForwarder.handle_info/2
backend  │     (stdlib) gen_server.erl:601: :gen_server.try_dispatch/4
backend  │     (stdlib) gen_server.erl:667: :gen_server.handle_msg/5
backend  │ Last message: :forward_interactions
```

**Comment by manukall** [comment]
hey @scottmessinger, this is quite hard to debug, so i have to ask a few questions:

can you please check what version of exprotobuf you have in `mix.lock`? 
do you have any idea where "role_grants" comes from? is this the name of an action or a channel event?
are you using custom instrumentation or custom traces?

**Comment by scottmessinger** [comment]
`role_grants` is the name of a collection in Mongo. 

Here are the dependencies:
```
  "exprotobuf": {:hex, :exprotobuf, "1.2.7", "287e2a1239fa9924c71815b1961e6e97d55c13496e9feb4ad755706af105ec36", [:mix], [{:gpb, "~> 3.24", [hex: :gpb, repo: "hexpm", optional: false]}], "hexpm"},
  "pryin": {:hex, :pryin, "1.2.0", "25cfa0cbeb4f7a03e929bad5180c6b895e19cc4a27eec2ac4e7dd519536ff102", [:mix], [{:ecto, "~> 2.0", [hex: :ecto, repo: "hexpm", optional: true]}, {:exprotobuf, "~> 1.2", [hex: :exprotobuf, repo: "hexpm", optional: false]}, {:hackney, "~> 1.2", [hex: :hackney, repo: "hexpm", optional: false]}, {:phoenix, "~> 1.2", [hex: :phoenix, repo: "hexpm", optional: true]}, {:plug, "~> 1.0", [hex: :plug, repo: "hexpm", optional: true]}, {:recon, "~> 2.3", [hex: :recon, repo: "hexpm", optional: false]}], "hexpm"},
```

**Comment by manukall** [comment]
@scottmessinger can you show me the code you use to instrument your mongodb calls? i can't find a way to reproduce that error.

**Comment by scottmessinger** [comment]
I'm not doing any special to instrument my Mongo calls -- I just followed the directions on the pryin site.

**Comment by manukall** [comment]
@scottmessinger could you give `resolve_query_functions` branch a try? i still couldn't reproduce the issue, but i hope this change could help.
also, could you tell me how you connect to mongo? pryin by default only instruments phoenix and ecto and i couldn't find a mongo adapter for ecto >= 2.0.

**Comment by manukall** [comment]
hey @scottmessinger, were you able to test this? i've just released version 1.2.1 with the above fix and another little one. could you give that (v1.2.1) a try, please?

**Comment by manukall** [comment]
@scottmessinger closing this for now. let me know if this is still an issue for you.
